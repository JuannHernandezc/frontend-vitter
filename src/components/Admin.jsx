import React from 'react'
import { useState,useEffect } from 'react'
import { auth } from '../helpers/firebase.js'
import { useNavigate } from "react-router";
import { Navbar } from './utils/Navbar.jsx';

export const Admin = () => {
  const [user, setUser] = useState(null);
  const navigate = useNavigate();
  useEffect(() => {
    if(auth.currentUser){
      setUser(auth.currentUser);
    }else{
      navigate('/');
    }
  },[navigate]);
  return (
    <div>
      <Navbar />
    </div>
  )
}
