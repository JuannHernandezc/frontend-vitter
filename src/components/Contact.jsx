import { Navbar } from "./utils/Navbar";
export const Contact = () => {
  return (
    <>
      <Navbar />
      <h1>Contact</h1>
    </>
  )
}
