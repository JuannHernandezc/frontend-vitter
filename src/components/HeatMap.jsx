import React from "react";
import { Map, TileLayer } from "react-leaflet";
import HeatmapLayer from "react-leaflet-heatmap-layer";
import "./styles/HeatMap.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faClose } from "@fortawesome/free-solid-svg-icons";

export const HeatMap = (locations) => {
  const position = [4.6126, -74.0705];
  const closedPopUp = () => {
    const popUp = document.getElementsByClassName("pop-up__heatmap");
    popUp[0].style.bottom = "125%";
  };
  return (
    <>
      <div className="container-close">
        <FontAwesomeIcon
          className="icon-close"
          icon={faClose}
          onClick={closedPopUp}
        />
      </div>
      <Map center={position} zoom={5} scrollWheelZoom={false}>
        <HeatmapLayer
          points={locations.data}
          longitudeExtractor={(m) => m.lng}
          latitudeExtractor={(m) => m.lat}
          intensityExtractor={(m) => parseFloat(m.lat)}
          max={100}
          minOpacity={0.6}
          blur={30}
        />
        <TileLayer
          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
      </Map>
    </>
  );
};
