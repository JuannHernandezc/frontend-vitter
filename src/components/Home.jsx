import { Footer } from "./utils/Footer";
import { Navbar } from "./utils/Navbar";
import { useNavigate } from "react-router";
import { auth } from "../helpers/firebase.js";
import "./styles/Home.css";
import { useEffect, useState } from "react";
export const Home = ({ firebaseUser }) => {
  const [user, setUser] = useState(null);
  const navigate = useNavigate();
  useEffect(() => {
    if (auth.currentUser) {
      setUser(auth.currentUser);
      navigate("/admin");
    } else {
      navigate("/");
    }
  }, [navigate]);
  const openPopUp = () => {
    const popUp = document.getElementsByClassName('pop-up__register');
    popUp[0].style.bottom = '0%';
  }
  return (
    <>
      <main className="main">
        <Navbar firebaseUser={firebaseUser} />
        <div className="container-sections">
          <section className="main__section animate__animated animate__fadeInTopLeft">
            <p>
              <span className="change-letter">Vitter</span> una nueva forma de
              mirar Twitter
            </p>
          </section>
          <section className="second-section">
            <p>Crear mapas de calor acerca de las tendencias de Twitter</p>
            <div className="second-section__picture">
              <img
                src="https://res.cloudinary.com/danhzm8qb/image/upload/v1653868164/vitter/mapaCalor_xsailf.jpg"
                alt="Mapa de calor"
              />
            </div>
            <button onClick={openPopUp} className="btn-create-map">Crear mapa</button>
          </section>
        </div>
        <Footer />
      </main>
    </>
  );
};
