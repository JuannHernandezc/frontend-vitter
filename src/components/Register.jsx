//Imports CSS
import "./styles/Register.css";

//Imports Components React

//Imports React
import { useState, useCallback } from "react";

//Imports FontAwesome
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUser, faClose } from "@fortawesome/free-solid-svg-icons";

//Imports React Router DOM
import { NavLink, useNavigate } from "react-router-dom";

//Imports Firebase
import { auth, db } from "../helpers/firebase.js";
import { createUserWithEmailAndPassword } from "firebase/auth";
import { doc, setDoc } from "firebase/firestore";

export const Register = () => {
  // Const Declarations

  const [name, setName] = useState("");
  const [lastName, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [phone, setPhone] = useState("");
  const [error, setError] = useState(null);
  const navigate = useNavigate();

  //Function that allow register firebase

  const register = useCallback(async () => {
    try {
      const answer = await createUserWithEmailAndPassword(
        auth,
        email,
        password
      );
      const col = await doc(db, "users", answer.user.uid);
      await setDoc(col, {
        uid: answer.user.uid,
        email: answer.user.email,
        name: name,
        lastName: lastName,
        phone: phone,
        type: "client",
      });
      setEmail("");
      setName("");
      setLastName("");
      setPassword("");
      setConfirmPassword("");
      setPhone("");
      setError(null);
      navigate("/admin");
    } catch (error) {
      if (error.code === "auth/invalid-email") {
        setError("Ingrese un Email valido");
      }
      if (error.code === "auth/email-already-in-use") {
        setError(
          "Este email ya se encuentra en uso, verifique si ya tiene una cuenta o cambielo"
        );
      }
    }
  }, [email, password, lastName, name, phone, navigate]);

  const processData = (e) => {
    e.preventDefault();
    if (!name.trim()) {
      setError("El campo Nombre no puede estar vacio");
      setTimeout(() => {
        setError(null);
      }, 15000);
      window.scrollTo(0, 0);
      return;
    }
    if (!lastName.trim()) {
      setError("El campo Apellido no puede estar vacio");
      setTimeout(() => {
        setError(null);
      }, 15000);
      window.scrollTo(0, 0);
      return;
    }
    if (!email.trim()) {
      setError("El campo Email no puede estar vacio");
      setTimeout(() => {
        setError(null);
      }, 15000);
      window.scrollTo(0, 0);
      return;
    }
    if (!password.trim()) {
      setError("El campo Contraseña no puede estar vacio");
      setTimeout(() => {
        setError(null);
      }, 15000);
      window.scrollTo(0, 0);
      return;
    }
    if (!confirmPassword.trim()) {
      setError("El campo Confirmar Contraseña no puede estar vacio");
      setTimeout(() => {
        setError(null);
      }, 15000);
      window.scrollTo(0, 0);
      return;
    }
    if(password.trim() !== confirmPassword.trim()){
      setError("El campo contraseña y confirmar contraseña no son iguales");
      setTimeout(() => {
        setError(null);
      }, 15000)
      window.scrollTo(0, 0);
      return;
    }
    if (!phone.trim()) {
      setError("El campo Numero Celular no puede estar vacio");
      setTimeout(() => {
        setError(null);
      }, 15000);
      window.scrollTo(0, 0);
      return;
    }
    register();
  };
  const closedPopUp = () => {
    const popUp = document.getElementsByClassName('pop-up__register');
    popUp[0].style.bottom = '125%';
  }
  return (
    <>
      <main className="container-main">
        <div className="container-close">
          <FontAwesomeIcon className="icon-close" icon={faClose} onClick={closedPopUp} />
        </div>
        <div className="container-icon">
          <h1>REGISTRO</h1>
        </div>
        {error && <p className="error">{error}</p>}
        <form className="container-form" onSubmit={processData}>
          <input
            type="text"
            placeholder="Nombre"
            onChange={(e) => setName(e.target.value)}
            value={name}
          />
          <input
            type="text"
            placeholder="Apellidos"
            onChange={(e) => setLastName(e.target.value)}
            value={lastName}
          />
          <input
            type="email"
            placeholder="Correo"
            onChange={(e) => setEmail(e.target.value)}
            value={email}
          />
          <input
            type="password"
            placeholder="Contraseña"
            onChange={(e) => setPassword(e.target.value)}
            value={password}
          />
          <input
            type="password"
            placeholder="Confirmar Contraseña"
            onChange={(e) => setConfirmPassword(e.target.value)}
            value={confirmPassword}
          />
          <input
            type="number"
            placeholder="Numero Celular"
            onChange={(e) => setPhone(e.target.value)}
            value={phone}
          />
          <NavLink className="form-item" to="/terms">
            Terminos y condiciones
          </NavLink>
          <button className="btn-register" type="submit">
            Registrarse
          </button>
        </form>
      </main>
    </>
  );
};
