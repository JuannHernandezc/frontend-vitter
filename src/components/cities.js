export const cityData = {
    "features": [
        {
          "city": "bogota", 
          "lat": 4.6126, 
          "lng": -74.0705, 
          "country": "colombia", 
          "iso2": "co", 
          "admin_name": "bogota", 
          "capital": "primary", 
          "population": "9464000", 
          "population_proper": "7963000"
        }

    ]
}