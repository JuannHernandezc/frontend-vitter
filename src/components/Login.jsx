//Imports CSS
import "./styles/Register.css";

//Imports FontAwesome
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowRightToBracket, faClose } from "@fortawesome/free-solid-svg-icons";

//Imports React Router DOM
import { NavLink, useNavigate } from "react-router-dom";

//Imports React js
import { useState, useCallback } from "react";

//Imports Firebase
import { signInWithEmailAndPassword } from "firebase/auth";
import { auth } from "../helpers/firebase.js";

export const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState("");
  const navigate = useNavigate();

  const login = useCallback(async () => {
    try {
      await signInWithEmailAndPassword(auth, email, password);
      setEmail("");
      setPassword("");
      setError(null);
      navigate("/admin");
    } catch (error) {
      if (error.code === "auth/invalid-email") {
        setError("Ingrese un Email valido");
      }
      if (error.code === "auth/user-not-found") {
        setError("Email NO registrado");
      }
      if (error.code === "auth/wrong-password") {
        setError("Contraseña incorrecta");
      }
    }
  }, [email, password, navigate]);

  const processData = (e) => {
    e.preventDefault();
    if (!email.trim()) {
      setError("El campo Email no puede estar vacio");
      setTimeout(() => {
        setError(null);
      }, 15000);
      window.scrollTo(0, 0);
      return;
    }
    if (!password.trim()) {
      setError("El campo Contraseña no puede estar vacio");
      setTimeout(() => {
        setError(null);
      }, 15000);
      window.scrollTo(0, 0);
      return;
    }
    login();
  };
  const closedPopUp = () => {
    const popUp = document.getElementsByClassName('pop-up__login');
    popUp[0].style.bottom = '125%';
  }
  return (
    <>
      <main className="container-login">
        <div className="container-close">
          <FontAwesomeIcon
            className="icon-close"
            icon={faClose}
            onClick={closedPopUp}
          />
        </div>
        <div className="container-icon">
          <FontAwesomeIcon className="icon" icon={faArrowRightToBracket} />
          <h1>Iniciar Sesión</h1>
        </div>
        {error && <p className="error">{error}</p>}
        <form className="container-form" onSubmit={processData}>
          <input
            type="email"
            placeholder="Correo"
            onChange={(e) => setEmail(e.target.value)}
            value={email}
          />
          <input
            type="password"
            placeholder="Contraseña"
            onChange={(e) => setPassword(e.target.value)}
            value={password}
          />
          <NavLink className="form-item" to="/rememberPassword">
            ¿Olvidaste tu contraseña?
          </NavLink>
          <button className="btn-login">Iniciar Sesión</button>
        </form>
      </main>
    </>
  );
};
