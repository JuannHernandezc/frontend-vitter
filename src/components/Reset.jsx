import React from 'react';
import "./styles/Register.css";

//Imports FontAwesome
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faKey, faClose } from "@fortawesome/free-solid-svg-icons";

//Imports React js
import { useState } from "react";

import { auth } from "../helpers/firebase.js";
import { sendPasswordResetEmail } from "@firebase/auth";
import { useCallback } from 'react';

export const Reset = () => {
    const [email, setEmail] = useState("");
    const [error, setError] = useState("");

    const resetPassword = useCallback( async () => {
      try {
        await sendPasswordResetEmail(auth,email);
        console.log("Correo Enviado")
      } catch (error) {
          setError(error.message);
      }
    },[email]) 
    const processData = (e) => {
      e.preventDefault();
      if (!email.trim()) {
        setError("El campo Email no puede estar vacio");
        setTimeout(() => {
          setError(null);
        }, 15000);
        window.scrollTo(0, 0);
        return;
      }
      resetPassword();
    };
    const closedPopUp = () => {
      const popUp = document.getElementsByClassName('pop-up__reset');
      popUp[0].style.bottom = '125%';
    }
  return (
    <main className="container-login">
    <div className="container-close">
      <FontAwesomeIcon
        className="icon-close"
        icon={faClose}
        onClick={closedPopUp}
      />
    </div>
    <div className="container-icon">
      <FontAwesomeIcon className="icon" icon={faKey} />
      <h1>Recuperar Contraseña</h1>
    </div>
    {error && <p className="error">{error}</p>}
    <form className="container-form" onSubmit={processData}>
      <input
        type="email"
        placeholder="Correo"
        onChange={(e) => setEmail(e.target.value)}
        value={email}
      />
      <button className="btn-reset">Recuperar Constraseña</button>
    </form>
  </main>
  )
}
