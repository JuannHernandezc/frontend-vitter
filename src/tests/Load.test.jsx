import { render } from "@testing-library/react";
import { Load } from "../components/Load";

describe( "Test in <TermsAndConditionsu />", () => {
    test("should do match with snapshot", () => {
        const {container} = render( <Load />);
        expect(container).toMatchSnapshot();
    });
});