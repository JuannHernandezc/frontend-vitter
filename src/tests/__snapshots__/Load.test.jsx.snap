// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <TermsAndConditionsu /> should do match with snapshot 1`] = `
<div>
  <div
    class="container-load"
  >
    <svg
      aria-busy="true"
      aria-label="blocks-loading"
      class="blocks-wrapper"
      data-testid="color-ring-svg"
      height="80"
      preserveAspectRatio="xMidYMid"
      role="status"
      viewBox="0 0 100 100"
      width="80"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <defs>
        <mask
          id="ldio-4offds5dlws-mask"
        >
          <circle
            cx="50"
            cy="50"
            r="26"
            stroke="#fff"
            stroke-dasharray="122.52211349000194 40.840704496667314"
            stroke-linecap="round"
            stroke-width="9"
            transform="rotate(198.018 50 50)"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              type="rotate"
              values="0 50 50;360 50 50"
            />
          </circle>
        </mask>
      </defs>
      <g
        mask="url(#ldio-4offds5dlws-mask)"
      >
        <rect
          fill="#e15b64"
          height="100"
          width="15"
          x="14.5"
          y="0"
        >
          <animate
            attributeName="fill"
            begin="-0.8s"
            dur="1s"
            keyTimes="0;0.25;0.5;0.75;1"
            repeatCount="indefinite"
            values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
          />
        </rect>
        <rect
          fill="#f47e60"
          height="100"
          width="15"
          x="28.5"
          y="0"
        >
          <animate
            attributeName="fill"
            begin="-0.6s"
            dur="1s"
            keyTimes="0;0.25;0.5;0.75;1"
            repeatCount="indefinite"
            values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
          />
        </rect>
        <rect
          fill="#f8b26a"
          height="100"
          width="15"
          x="42.5"
          y="0"
        >
          <animate
            attributeName="fill"
            begin="-0.4s"
            dur="1s"
            keyTimes="0;0.25;0.5;0.75;1"
            repeatCount="indefinite"
            values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
          />
        </rect>
        <rect
          fill="#abbd81"
          height="100"
          width="15"
          x="56.5"
          y="0"
        >
          <animate
            attributeName="fill"
            begin="-0.2s"
            dur="1s"
            keyTimes="0;0.25;0.5;0.75;1"
            repeatCount="indefinite"
            values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
          />
        </rect>
        <rect
          fill="#849b87"
          height="100"
          width="15"
          x="70.5"
          y="0"
        >
          <animate
            attributeName="fill"
            begin="0s"
            dur="1s"
            keyTimes="0;0.25;0.5;0.75;1"
            repeatCount="indefinite"
            values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
          />
        </rect>
      </g>
    </svg>
  </div>
</div>
`;
